src/components/tl/ucp/alltoall/alltoall_pairwise.c

```diff
@@ -75,7 +48,7 @@ void ucc_tl_ucp_alltoall_pairwise_progress(ucc_coll_task_t *coll_task)
                ((task->tagged.recv_posted - task->tagged.recv_completed) <
                 nreqs)) {
             peer = get_recv_peer(grank, gsize, task->tagged.recv_posted);
-            UCPCHECK_GOTO(ucc_tl_ucp_recv_nb((void *)(rbuf + peer * data_size),
+            UCPCHECK_GOTO(ucc_tl_ucp_recv_nb(PTR_OFFSET(rbuf, peer * data_size),
```

Comment:
if `ucc_tl_ucp_recv_nb` returns **UCC_INPROGRESS** it jumps to out, but you could still post more tasks to match nreq tasks in flight